src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -457,7 +465,15 @@ private function validateClassNode($object, $cacheKey, ClassMetadataInterface $m
             $defaultOverridden = false;
 
             // Use the object hash for group sequences
-            $groupHash = \is_object($group) ? spl_object_hash($group) : $group;
+            if (\is_object($group)) {
+                $groupHash = spl_object_hash($group);
+
+                if (!isset($this->validatedObjectsReferences[$groupHash])) {
```

Comment:
Why is this needed for group sequences?

Reply:
GroupSequences can suffer from the same problem since they can be passed directly to the validate method as objects.